### ISSUE TITLE
Migrate IRC bot to c:geo server

### ISSUE THREAD

**Issue by SammysHP** [cgeo/cgeo#9033]
Our IRC bot is currently running on a server of my former university. Sometimes it requires a restart after a netsplit because it doesn't detect what is going on.

I'd like to migrate the bot to our c:geo server. It requires a Java environment (currently JRE 8) and Scala (currently 2.11). It is a single jar container, a configuration file and a startup script which is called periodically by a cron job (could be migrated to a systemd user service). Last touched 2014.

There is not much activity in our IRC channel anymore (I think 2-5 messages a month) so it might not hurt to disable it completely. But in any case I'd like to get rid of it on this server as I don't know how long I have access to it.

**Comment by mucek4** [comment]
Do we have an option to make a Docker image for thisd bot?

**Comment by Lineflyer** [comment]
@mucek4 In a discussion thread I suggested to start using IRC again for better efficiency. In order for that to make more sense the IRC bot which logs the conversation would again have more importance ...not to say its kind of mandatory. Any chance we can move it?

**Comment by mucek4** [comment]
If @SammysHP makes a docker image thet would be great.

**Comment by SammysHP** [comment]
Working on it.

**Comment by SammysHP** [comment]
@mucek4
Here is all you need.

https://cloud.finf.uni-hannover.de/s/f246gZWALEByqTC

After building mount the log directory into /app/logs. I used this command for testing:

```
docker run --rm -v /some/directory/:/app/logs/ cgeo-irct
```

**Comment by mucek4** [comment]
Downloaded. Will get it up & running

**Comment by SammysHP** [comment]
If it works I will send you an archive with all previous logs and will shut down my instance.

**Comment by SammysHP** [comment]
Just noticed, that I already have a cron that archives all logs. :D 

https://krul.finf.uni-hannover.de/~sammyshp/cgeo-logs.tar.gz

@mucek4 My bot is stopped and you can download the logs from the link above.

**Comment by mucek4** [comment]
I have downloaded all logs one by one before you gave me link. Currently I have just started bot and there is no link for logs from web. Could you run the bot for a day or two so I connect everything?

**Comment by SammysHP** [comment]
OK, I've restarted it. :)

**Comment by SammysHP** [comment]
@mucek4 I've disabled my instance again because two instances don't interact well with each other.

**Comment by Lineflyer** [comment]
FYI:
https://irclogs.cgeo.org seems to still show the old bot IMHO.

**Comment by mucek4** [comment]
@Lineflyer I know. That's the reason I said @SammysHP shall run its bot in paralel for some time. I just started the bot. I yet have to set backups, log redirection to web, make it persistent, create notification if it crashes, etc...

**Comment by SammysHP** [comment]
But we can't run it in parallel, that caused some issues.

**Comment by Lineflyer** [comment]
@mucek4 

Any update with the IRC bot?
Would be nice to have the log available again in order to have a meaningful channel usage option (so anyone can look up the history he might have missed).

**Comment by SammysHP** [comment]
Btw

> […] create notification if it crashes

I've never seen it crash. It just starts to idle, thinking it's still connected after a netsplit sometimes.

**Comment by eddiemuc** [comment]
> @mucek4
> 
> Any update with the IRC bot?
> Would be nice to have the log available again in order to have a meaningful channel usage option (so anyone can look up the history he might have missed).

I can really recommend usage of https://element.io/ in combination with https://matrix.org/docs/projects/try-matrix-now/. Then the history is simply stored on your phone (like with other messengers)

**Comment by mucek4** [comment]
Logs are back. Except URL is not public accessible, I still have to find out how to export plain text version.

**Comment by mucek4** [comment]
I have fixed that too.
